src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -997,13 +997,23 @@ public boolean entityInteract(GlowPlayer player, InteractEntityMessage message)
         super.entityInteract(player, message);
 
         if (message.getHandSlot() == EquipmentSlot.HAND) {
-            if (!InventoryUtil.isEmpty(player.getItemInHand()) && player.getItemInHand().getType() == Material.LEASH) {
+            ItemStack handUten = player.getItemInHand();
```

Comment:
It appears you can use the off-hand with leads, and `getItemInHand()` actually gets the main hand.

Also, you are calling the leashing event without checking if the player is 'allowed' to do it with that entity type. I think the type check should also be done here.